Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -297,12 +297,28 @@ public class CryptoStore: ObservableObject {
       if isInitialOpen {
         portfolioStore.discoverAssetsOnAllSupportedChains()
       }
+      
       let pendingTransactions = await fetchPendingTransactions()
       var newPendingRequest: PendingRequest?
       if !pendingTransactions.isEmpty {
         newPendingRequest = .transactions(pendingTransactions)
-      } else { // no pending transactions, check for webpage requests
-        newPendingRequest = await fetchPendingWebpageRequest()
+      } else {
+        // no pending transactions, but need to check if
+        // `TransactionConfirmationView` is ready to be dismissed
+        if let store = confirmationStore, !store.isReadyToBeDismissed {
+          /*
+           We need to check if Tx Confirmation modal is ready
+           to be dismissed. It could be not ready as there is no pending tx
+           but an active tx is being shown its different state like
+           loading, submitted, completed or failed.
+           As such we need to continue displaying Tx Confirmation until the
+           user taps Ok/Close on the status overlay.
+           */
+          newPendingRequest = .transactions([])
+        } else {
+          // check for webpage requests
+          newPendingRequest = await fetchPendingWebpageRequest()
+        }
       }
```

Comment:
nit, this doesn't need nested:
```suggestion
      } else if let store = confirmationStore, !store.isReadyToBeDismissed {
        /*
         We need to check if Tx Confirmation modal is ready
         to be dismissed. It could be not ready as there is no pending tx
         but an active tx is being shown its different state like
         loading, submitted, completed or failed.
         As such we need to continue displaying Tx Confirmation until the
         user taps Ok/Close on the status overlay.
         */
        newPendingRequest = .transactions([])
      } else {
        // check for webpage requests
        newPendingRequest = await fetchPendingWebpageRequest()
      }
```

Reply:
updated in [8d59b95](https://github.com/brave/brave-ios/pull/6960/commits/8d59b95439fdfcd0da5f733c63d51fea80b2a899)